src/main/java/org/pac4j/j2e/util/Pac4JHttpServletRequestWrapper.java

```diff
@@ -0,0 +1,51 @@
+package org.pac4j.j2e.util;
+
+import java.security.Principal;
+import java.util.Collection;
+import java.util.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileHelper;
+
+/**
+ * This wraps the {@link HttpServletRequest} to make it pac4j-aware by with respect to user-related methods.
+ * 
+ * @author Victor Noel
+ * @since 4.0.0
+ *
+ */
+public class Pac4JHttpServletRequestWrapper extends HttpServletRequestWrapper {
+
+    private Collection<CommonProfile> profiles;
+
+    public Pac4JHttpServletRequestWrapper(HttpServletRequest request, Collection<CommonProfile> profiles) {
+        super(request);
+        this.profiles = profiles;
+    }
+
+    @Override
+    public String getRemoteUser() {
+        return getPrincipal().map(p -> p.getName()).orElse(null);
```

Comment:
You removed the identifier as a fallback if the username is blank? I thought it was better the other way...

Reply:
It's because our implementation of Principal already those this for us :)

Reply:
You're right. Good point.